### PR TITLE
fix: cloudfront response logging.

### PR DESCRIPTION
**Pull request by Bento007**
### Reviewers
**Functional:** @nayib-jose-gloria 

**Readability:** @ebezzi 

---

## Changes
- fix how we are logging the cloudfront invalidation response.
- Catch and log all exceptions and raise internal error on API.

### CODE REVIEW

File: backend/corpora/common/utils/cloudfront.py [hunk 1]
```diff
@@ -34,7 +34,7 @@ def _create_invalidation(distribution: str, paths: List[str]):
             "CallerReference": invalidation_id,
         },
     )
-    logging.info(**response.json, status_code=response.status_code)
+    logging.info(dict(body=response.json, status_code=response.status_code))
```
Comment (danieljhegeman) [inline backend/corpora/common/utils/cloudfront.py:37]: this was the issue?

File: backend/corpora/api_server/app.py [hunk 2]
```diff
@@ -174,7 +172,7 @@ def handle_corpora_error(exception):
     )
 
 
-@app.errorhandler(InternalServerError)
+@app.errorhandler(Exception)
```
Comment (danieljhegeman) [inline backend/corpora/api_server/app.py:175]: Isn't this also going to be triggered by exceptions that produce responses for the client, such as 400 responses, and which are not server errors? 🤔

**Review by danieljhegeman — COMMENTED** [review]
see q about catching `Exception`

File: backend/corpora/api_server/app.py (hunk 2 above)
Comment (Bento007) [reply backend/corpora/api_server/app.py:175]: The 400 responses for invalid parameters are thrown by connexion which throws a ProblemException. We have a handler for ProblemExceptions above.

File: backend/corpora/common/utils/cloudfront.py (hunk 1 above)
Comment (Bento007) [reply backend/corpora/common/utils/cloudfront.py:37]: Yes

File: backend/corpora/api_server/app.py (hunk 2 above)
Comment (Bento007) [reply backend/corpora/api_server/app.py:175]: We would be failing a bunch of unit tests if this wasn't working properly.

Comment (danieljhegeman) [reply backend/corpora/api_server/app.py:175]: Ok. As long as we're sure we aren't masking any non-500 errors with this.

**Review by danieljhegeman — APPROVED** [review]

**Pull request merged**